oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/IndexStructures.java

```diff
@@ -146,4 +172,18 @@ private Map<String, Object> diffFields(Fields fields) {
                                   ));
         }
     }
+
+    /**
+     * The index settings structure which only include needs to compare and update fields
+     */
+    @EqualsAndHashCode
+    public static class UpdatableIndexSettings {
+        private final int replicas;
+        private final int shards;
+
+        public UpdatableIndexSettings(Map<String, Object> indexSettings) {
+            this.replicas = Integer.parseInt((String) indexSettings.getOrDefault("number_of_replicas", "0"));
+            this.shards = Integer.parseInt((String) indexSettings.getOrDefault("number_of_shards", "0"));
```

Comment:
I am fully aware how it works globally, but I have some doubts.

1. Is it mandatory to set always both properties for a given index in the json to specify the concrete number of replicas and shards?
2. If only replicas is provided for instance, what will be the number of shards 0? Or the value specified by `SW_STORAGE_ES_INDEX_SHARDS_NUMBER` if provided or the default value of `SW_STORAGE_ES_INDEX_SHARDS_NUMBER` if absent?

Reply:
Thanks, I'll check.

Reply:
> 1. Is it mandatory to set always both properties for a given index in the json to specify the concrete number of replicas and shards?

No, we can set one or both, the absent one will use the generic settings.
see: https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-storage/#index-settings

> 2. If only replicas is provided for instance, what will be the number of shards 0? 

No, `UpdatableIndexSettings` structure only used for comparing old and new settings, so the the number of shards will be the the generic settings which I mentioned above.
If 
```java
Integer.parseInt((String) indexSettings.getOrDefault("number_of_replicas", "0"));
```
 cause some confusion, I'll submit a PR, and change it to 
```java
String.valueOf(indexSettings.getOrDefault("number_of_replicas", Const.EMPTY_STRING));
```

>Or the value specified by `SW_STORAGE_ES_INDEX_SHARDS_NUMBER` if provided or the default value of `SW_STORAGE_ES_INDEX_SHARDS_NUMBER` if absent?

yes, they have a default value
see: https://github.com/apache/skywalking/blob/48a0985345bc65a63b17d4539f80de334383dedb/oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/StorageModuleElasticsearchConfig.java#L60